Classes/Hook/FrontendEditingInitializationHook.php

```diff
@@ -466,15 +471,96 @@ protected function getAbsolutePath(string $path): string
      */
     protected function getPageTreeStructure(): array
     {
+        $entryPoints = $this->getAllEntryPointPageTrees();
+
+        foreach ($entryPoints as $entryPoint) {
+            if ($entryPoint['uid'] == 0) {
+                $children = $this->getStructureForSinglePageTree($this->typoScriptFrontendController->rootLine[0]['uid']);
+            } else {
+                $children[] = $this->getStructureForSinglePageTree($entryPoint['uid'])[0];
+            }
+        }
+
         return [
             'name' => $GLOBALS['TYPO3_CONF_VARS']['SYS']['sitename'],
             'icon' => $this->getAbsolutePath('EXT:core/Resources/Public/Icons/T3Icons/apps/apps-pagetree-root.svg'),
-            'children' => $this->getStructureForSinglePageTree(
-                $this->typoScriptFrontendController->rootLine[0]['uid']
-            )
+            'children' => $children
         ];
     }
 
+    /**
+     * Fetches all entry points for the page tree that the user is allowed to see
+     * This was taken from class TreeController (namespace TYPO3\CMS\Backend\Controller\Page);
+     *
+     * @return array
+     */
+    protected function getAllEntryPointPageTrees(): array
+    {
+        $backendUser = $GLOBALS['BE_USER'];
+
+        $userTsConfig = $GLOBALS['BE_USER']->getTSConfig();
+        $excludedDocumentTypes = GeneralUtility::intExplode(',', $userTsConfig['options.']['pageTree.']['excludeDoktypes'] ?? '', true);
+
+        $additionalPageTreeQueryRestrictions = [];
+        if (!empty($excludedDocumentTypes)) {
+            foreach ($excludedDocumentTypes as $excludedDocumentType) {
+                $additionalPageTreeQueryRestrictions[] = new DocumentTypeExclusionRestriction((int)$excludedDocumentType);
+            }
+        }
+
+        $repository = GeneralUtility::makeInstance(PageTreeRepository::class, (int)$backendUser->workspace, [], $additionalPageTreeQueryRestrictions);
+
+        $entryPoints = (int)($backendUser->uc['pageTree_temporaryMountPoint'] ?? 0);
+        if ($entryPoints > 0) {
+            $entryPoints = [$entryPoints];
+        } else {
+            $entryPoints = array_map('intval', $backendUser->returnWebmounts());
+            $entryPoints = array_unique($entryPoints);
+            if (empty($entryPoints)) {
+                // use a virtual root
+                // the real mount points will be fetched in getNodes() then
+                // since those will be the "sub pages" of the virtual root
+                $entryPoints = [0];
+            }
+        }
+        if (empty($entryPoints)) {
+            return [];
+        }
+
+        foreach ($entryPoints as $k => &$entryPoint) {
+            /*
```

Comment:
Remove the commented code.

Reply:
The reason, why I commented this out was, that I don't have the data for `$this->hiddenRecords`, but I was not sure, if this is somehow needed... What to do? Just remove it?

Reply:
One more information on this: As stated in the comments to the method `getAllEntryPointPageTrees()`, the code was taken form the core.

Reply:
@MattiasNilsson Checking the code today again, I was able to fix this.